Bio_Seq_Analysis_Tool.py

```diff
@@ -0,0 +1,163 @@
+                'reverse_complement': na.reverse_complement,
+                'complement': na.complement,
+                'gc_calculate': na.gc_calculate}
+    analysis = []
+    for seq in args:
+        na.is_na(seq)
+        if operation in OPERATION_DICT.keys():
+            analysis.append(OPERATION_DICT[operation](seq))
+        else:
+            raise ValueError(f'Wrong operation!')
+    return analysis
+
+
+def analyse_fastq(input_path: str, 
+                  gc_bounds: Union[int, float, Tuple [int], Tuple [float]] = (0, 100), 
+                  length_bounds: Union[int, Tuple [int]] = (0, 2**32),
+                  quality_threshold: float = 0.0, filtered_file_name: Union[None, str] = None) -> Dict[str,str]:
+    """
+    This function help analyze a set of reads obtained from next-generation sequencing. 
+    
+    The function allow to filter the desired reads according to three parameters:
+    GC-content, length and reading quality.
+    
+    :param seqs: 
+    Path to the file with FASTQ-sequences in the format. 
+    :type seqs: str
+    
+    :param gc_bounds: 
+    Boundary parameters for filtering sequences by GC-content. Save only reads with a GC-content between boundaries 
+    or lower than one boundary. Lower boundary cannot be less than 0 and upper boundary cannot be greater than 100. 
+    gc_bounds default value is (0,100).
+    :type param gc_bounds: Union[int, float, Tuple [int], Tuple [float]]
+    
+    :param length_bounds: 
+    Boundary parameters for filtering sequences by length. Works the same as gc_bounds. Lower boundary cannot be less 
+    than 0 and upper boundary cannot be greater than 2^32. length_bounds default value is (0,2^32)
+    :type param length_bounds: Union[int, Tuple [int]
+    
+    :param quality_threshold: 
+    Threshold for quality of each nucleotide in read. Quality incodes by ASCII codes. The threshold cannot be more 
+    than 40. quality_threshold default value is 0 
+    :type param quality_threshold: float
+    
+    :return: 
+    New dictionaries with fastq sequence.The first one consisting of filtered fastq sequences and the other one with 
+    sequences that did not pass filters.
+    :rtype: Dict[str]
+    
+    :raises ValueError: if sequence not RNA or DNA, also if the argument values are outside the allowed ones
+    """
+    seqs, path_to_file = fq.read_fastq(input_path)
+    file_name = path_to_file.split("/")[-1]
+    if type(gc_bounds) == float or type(gc_bounds) == int:
+        gc_bounds = (0,gc_bounds)
+        if gc_bounds[0] < 0 or gc_bounds[1] > 100:
+            raise ValueError(f'Wrong boundaries!')
+    if type(length_bounds) == int:
+        length_bounds = (0,length_bounds)
+        if length_bounds[0] < 0 or length_bounds[1] > 2**32:
+            raise ValueError(f'Wrong boundaries!')        
+    if quality_threshold > 40:
+        raise ValueError(f'Wrong quality threshold!')
+    analysed_seq = {}
+    error_seq = {}
+    for seq in seqs.items():
```

Comment:
Там потом у тебя обращения в духе `seq[0][1]`. Так сложно понять что  хранится в seq. Можно сперва распокавать в переменные с осмысленными названиями. В целом хорошо это сделать внутри цикла в начале по индексам, но я покажу как можно сделать прямо в цикле:
```suggestion
    for name, (seq, comment, qual) in  seqs.items():
```